### PR TITLE
Quality: Parameterize SDK and test-data branches in CI workflow

### PR DESCRIPTION
**Pull request by typotter**
🎟️ Fixes FF-3141
🎟️ Fixes FF-3095
 towards FF-3085

👯‍♂️ **Related PRs**
- [sdk-test-data#81](https://github.com/Eppo-exp/sdk-test-data/pull/81)

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
_This change_
🚀 - the CI  workflow is enhanced into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), exposing parameters for SDK branch and the sdk-test-data branch to use in testing.
- The test workflow runs using the main branch of sdk-test-data on all main pushes and Pull Requests using the PR's branch

_External to this Change_
[sdk-test-data](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows) get two testing workflows.
1. ♻️  "Local Testing"- For all pull request changes, the "Local Testing" workflow calls the reusable SDK workflows, test results are recorded only in the sdk-test-data repository. This is run using the main SDK branch and the "current" branch of workflow, i.e. the pull request branch. (SDK repo does not see/is not notified of test failures during PR lifecycle, only on push to main)
2. ♻️ 🧑‍💻 "Remote Testing" - On all pushes to the main branch of sdk-test-data, the SDK testing workflows are triggered to run within their respective repositories, alerting all subscribers of failed runs (repo is red/green).

### CODE REVIEW

**Review by leoromanovsky — COMMENTED** [review]
There are some test failures related to file paths. Can I help here?

**Comment by typotter** [comment]
> There are some test failures related to file paths. Can I help here?

Thanks. I dropped cloning the test data for python and ruby since they didn't have universal tests but I missed that they're got a couple static tests using the data files.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,7 +33,12 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          submodules: true
+          repository: Eppo-exp/eppo-multiplatform
```
Comment (leoromanovsky) [inline .github/workflows/ci.yml:36]: This one is not an SDK - it is a standalone server that uses `eppo-core` so it won't have the same evaluation testing criteria. Ignore it or another idea to handle?

**Review by leoromanovsky — APPROVED** [review]
python will get fixed in https://github.com/Eppo-exp/eppo-multiplatform/pull/77

File: sdk-test-data
Comment (rasendubi) [inline sdk-test-data:1]: Would like to keep this as a submodule for all the reasons mentioned in https://github.com/Eppo-exp/eppo-multiplatform/pull/22

**Review by rasendubi — DISMISSED** [review]

**Review by leoromanovsky — APPROVED** [review]
